retrofit-adapters/rxjava/src/main/java/retrofit/RxJavaCallAdapterFactory.java

```diff
@@ -110,8 +112,9 @@ private CallOnSubscribe(Call<T> originalCall) {
           try {
             subscriber.onNext(response);
           } catch (Throwable t) {
+            Exceptions.throwIfFatal(t);
```

Comment:
Actually I lied. It only eats IOExceptions. These will just kill the executor thread.